vector/src/main/java/fr/gouv/tchap/util/DinsicUtils.java

```diff
@@ -101,32 +101,93 @@ public  static boolean isFromFrenchGov(final List<String> emails) {
      */
-    public static @Nullable String getNameFromDisplayName(@Nullable String displayName) {
+    @Nullable
+    public static String getNameFromDisplayName(@Nullable String displayName) {
         String myRet = displayName;
         if (null != displayName && displayName.contains("[")) {
             myRet = displayName.split("\\[")[0].trim();
         }
         return myRet;
     }
+
     /**
      * Get the potential domain name from a display name.
      * For example in case of "Jean Martin [Modernisation]", this will return "Modernisation".
      *
      * @param displayName
-     * @return displayName without name, empty string if no domain is available.
+     * @return displayName without name, null if no domain is available.
      */
+    @Nullable
     public static String getDomainFromDisplayName(@Nullable String displayName) {
-        String myRet = "";
-
+        String myRet = null;
         if (null != displayName && displayName.contains("[")) {
             myRet = displayName.split("\\[")[1];
             if (myRet.contains("]")) {
-                myRet = myRet.split("\\]")[0];
+                myRet = myRet.split("\\]")[0].trim();
             } else {
-                myRet = "";
+                myRet = null;
+            }
+        }
+        return myRet;
+    }
+
+    /**
+     * Get a display name from the tchap user identifier.
+     * We don't extract the domain for the moment in order to not display unexpected information.
+     * For example in case of "@jean.martin-modernisation.fr:matrix.org", this will return "Jean Martin".
+     *
+     * @param tchapUserId user id
+     * @return displayName without domain, null if the id is not valid.
+     */
+    @Nullable
+    public static String getDisplayNameFromUserId(@Nullable String tchapUserId) {
+        String displayName = null;
+
+        if (null != tchapUserId && MXSession.PATTERN_CONTAIN_MATRIX_USER_IDENTIFIER.matcher(tchapUserId).matches()) {
+            int index = tchapUserId.indexOf("-");
+            if (-1 != index) {
+                // Retrieve the user name by ignoring the first character '@' of the matrix id
+                displayName = tchapUserId.substring(1, index);
+                String[] components = displayName.split("\\.");
+
+                StringBuilder builder = new StringBuilder();
+                for (String component : components) {
+                    if (component.length() > 0) {
+                        String updatedComponent = component.substring(0, 1).toUpperCase();
+                        if (component.length() > 1) {
+                            updatedComponent += component.substring(1);
+                        }
+
+                        if (builder.toString().isEmpty()) {
+                            builder.append(updatedComponent);
+                        } else {
+                            builder.append(" " + updatedComponent);
```

Comment:
It's not the end of the world, but we use StringBuilders because it is more efficient than concatenating String with `+`. So it's better to write
`builder.append(" ").append(updatedComponent);`